shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm

```diff
@@ -24,124 +44,129 @@
-#else   // TARGET_IPHONE_SIMULATOR
-  fml::scoped_nsobject<NSNetService> _netService;
-#endif  // TARGET_IPHONE_SIMULATOR
+@interface ObservatoryNSNetServiceDelegate
+    : NSObject <FlutterObservatoryPublisherDelegate, NSNetServiceDelegate>
+@end
 
-  flutter::DartServiceIsolate::CallbackHandle _callbackHandle;
-  std::unique_ptr<fml::WeakPtrFactory<FlutterObservatoryPublisher>> _weakFactory;
-}
+@interface ObservatoryDNSServiceDelegate : NSObject <FlutterObservatoryPublisherDelegate>
+@end
 
-- (NSURL*)url {
-  return _url.get();
+@implementation ObservatoryDNSServiceDelegate {
+  fml::scoped_nsobject<FlutterObservatoryPublisher> _owner;
+  DNSServiceRef _dnsServiceRef;
 }
 
-- (instancetype)init {
+@synthesize url;
+
+- (instancetype)initWithOwner:(FlutterObservatoryPublisher*)owner {
   self = [super init];
   NSAssert(self, @"Super must not return null on init.");
-
-  _weakFactory = std::make_unique<fml::WeakPtrFactory<FlutterObservatoryPublisher>>(self);
-
-  fml::MessageLoop::EnsureInitializedForCurrentThread();
-
-  _callbackHandle = flutter::DartServiceIsolate::AddServerStatusCallback(
-      [weak = _weakFactory->GetWeakPtr(),
-       runner = fml::MessageLoop::GetCurrent().GetTaskRunner()](const std::string& uri) {
-        runner->PostTask([weak, uri]() {
-          if (weak) {
-            [weak.get() publishServiceProtocolPort:std::move(uri)];
-          }
-        });
-      });
-
+  _owner.reset([owner retain]);
   return self;
 }
 
 - (void)stopService {
-#if TARGET_IPHONE_SIMULATOR
   if (_dnsServiceRef) {
     DNSServiceRefDeallocate(_dnsServiceRef);
     _dnsServiceRef = NULL;
   }
-#else   // TARGET_IPHONE_SIMULATOR
-  [_netService.get() stop];
-  [_netService.get() setDelegate:nil];
-#endif  // TARGET_IPHONE_SIMULATOR
-}
-
-- (void)dealloc {
-  [self stopService];
-
-  flutter::DartServiceIsolate::RemoveServerStatusCallback(std::move(_callbackHandle));
-  [super dealloc];
 }
 
-- (void)publishServiceProtocolPort:(std::string)uri {
-  [self stopService];
-  if (uri.empty()) {
-    return;
-  }
+- (void)publishServiceProtocolPort:(NSString*)uri {
   // uri comes in as something like 'http://127.0.0.1:XXXXX/' where XXXXX is the port
   // number.
-  _url.reset([[NSURL alloc] initWithString:[NSString stringWithUTF8String:uri.c_str()]]);
-
-  NSString* serviceName =
-      [[[NSBundle mainBundle] infoDictionary] objectForKey:@"CFBundleIdentifier"];
-
-  // Check to see if there's an authentication code. If there is, we'll provide
-  // it as a txt record so flutter tools can establish a connection.
-  auto path = std::string{[[_url path] UTF8String]};
-  if (!path.empty()) {
-    // Remove leading "/"
-    path = path.substr(1);
-  }
-  NSData* pathData = [[[NSData alloc] initWithBytes:path.c_str() length:path.length()] autorelease];
-  NSDictionary* txtDict = @{
-    @"authCode" : pathData,
-  };
-  NSData* txtData = [NSNetService dataFromTXTRecordDictionary:txtDict];
+  url.reset([[[NSURL alloc] initWithString:uri] retain]);
```

Comment:
`self.url`, or am I missing where this `url` is coming from?

Reply:
Done

Reply:
Undone haha.

`url` is the synethsized property from the protocol right?